src/server/pps/server/api_server.go

```diff
@@ -1566,12 +1566,26 @@ func (a *apiServer) jobManager(ctx context.Context, jobInfo *pps.JobInfo) {
 		updateProgress(0)
 
 		serviceAddr, err := a.workerServiceIP(ctx, rcName)
+		conns := make(map[*grpc.ClientConn]bool)
+		var connsMu sync.Mutex
+		defer func() {
+			for conn := range conns {
```

Comment:
Hmm, I don't think it's necessary here because that's guaranteed to be called after the pool is no longer in use. We'd have some other big problems if that wasn't the case since we'd be closing connections that had just been returned and might be in use.